flink-runtime/src/main/java/org/apache/flink/runtime/messages/webmonitor/JobStatusesWithIdsOverview.java

```diff
@@ -0,0 +1,368 @@
+		this.jobsReconciling = combine(first.getJobsReconciling(), second.getJobsReconciling());
+	}
+
+	public List<JobID> getJobsCreated() {
+		return jobsCreated;
+	}
+
+	public List<JobID> getJobsRunningOrPending() {
+		return jobsRunningOrPending;
+	}
+
+	public List<JobID> getJobsFinished() {
+		return jobsFinished;
+	}
+
+	public List<JobID> getJobsCancelling() {
+		return jobsCancelling;
+	}
+
+	public List<JobID> getJobsCancelled() {
+		return jobsCancelled;
+	}
+
+	public List<JobID> getJobsFailing() {
+		return jobsFailing;
+	}
+
+	public List<JobID> getJobsFailed() {
+		return jobsFailed;
+	}
+
+	public List<JobID> getJobsRestarting() {
+		return jobsRestarting;
+	}
+
+	public List<JobID> getJobsSuspended() {
+		return jobsSuspended;
+	}
+
+	public List<JobID> getJobsReconciling() {
+		return jobsReconciling;
+	}
+
+	// ------------------------------------------------------------------------
+
+
+	@Override
+	public int hashCode() {
+		return jobsCreated.hashCode() ^
+				jobsRunningOrPending.hashCode() ^
+				jobsFinished.hashCode() ^
+				jobsCancelling.hashCode() ^
+				jobsCancelled.hashCode() ^
+				jobsFailing.hashCode() ^
+				jobsFailed.hashCode() ^
+				jobsRestarting.hashCode() ^
+				jobsSuspended.hashCode() ^
+				jobsReconciling.hashCode();
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (obj == this) {
+			return true;
+		}
+		else if (obj instanceof JobStatusesWithIdsOverview) {
+			JobStatusesWithIdsOverview that = (JobStatusesWithIdsOverview) obj;
+			return this.jobsCreated.equals(that.jobsCreated) &&
+					this.jobsRunningOrPending.equals(that.jobsRunningOrPending) &&
+					this.jobsFinished.equals(that.jobsFinished) &&
+					this.jobsCancelling.equals(that.jobsCancelling) &&
+					this.jobsCancelled.equals(that.jobsCancelled) &&
+					this.jobsFailing.equals(that.jobsFailing) &&
+					this.jobsFailed.equals(that.jobsFailed) &&
+					this.jobsRestarting.equals(that.jobsRestarting) &&
+					this.jobsSuspended.equals(that.jobsSuspended) &&
+					this.jobsReconciling.equals(that.jobsReconciling);
+		}
+		else {
+			return false;
+		}
+	}
+
+	@Override
+	public String toString() {
+		return "JobStatusesWithIdsOverview {" +
+				"createdJobs=" + jobsCreated +
+				", runningOrPendingJobs=" + jobsRunningOrPending +
+				", finishedJobs=" + jobsFinished +
+				", cancellingJobs=" + jobsCancelling +
+				", cancelledJobs=" + jobsCancelled +
+				", failingJobs=" + jobsFailing +
+				", failedJobs=" + jobsFailed +
+				", restartingJobs=" + jobsRestarting +
+				", suspendedJobs=" + jobsSuspended +
+				", reconcilingJobs=" + jobsReconciling +
+				'}';
+	}
+
+	// ------------------------------------------------------------------------
+	//  Message serializers
+	// ------------------------------------------------------------------------
+
+	public static final class JobStatusesWithIdsOverviewSerializer extends StdSerializer<JobStatusesWithIdsOverview> {
```

Comment:
Just out of curiosity, why did you define a (de)serializer for this class? The structure looks as if this could have been done automatically by the `ObjectMapper`.

Reply:
I think it was due to the serialization of `JobID`s. Maybe what should be enough is to define a (de)serializer for the `JobID` class?

Reply:
Yes, then let's define a `JobID(De)Serializer` similar to the `JobVertexIDSerializer` and annote the respective field.

Reply:
👍 will change